Dockerfiles/CI-Dockerfile

```diff
@@ -0,0 +1,43 @@
+### Note: This DockerFile is adapted from https://github.com/ucb-bar/chipyard/blob/master/.circleci/images/Dockerfile which was adapted from: https://github.com/CircleCI-Public/example-images/openjdk
+# This is a chipyard setup for CI, no toolchains are initialized
```

Comment:
Now that we know this mainly works. Can we dedup this Dockerfile with the other one so that the only difference is the `init-submodules` and `build-toolchains` lines? Something like this: https://github.com/QuarantineGemmini/Dockerfiles